indico/modules/events/registration/clone.py

```diff
@@ -73,8 +76,17 @@ def _clone_form_items(self, old_form, new_form, clone_all_revisions):
                         new_item.current_data = field_data
                         self._field_data_map[old_item.current_data] = field_data
                 new_section.children.append(new_item)
+                item_map[old_item.id] = new_item
+                db.session.add(new_item)
             new_form.form_items.append(new_section)
+            db.session.add(new_section)
             db.session.flush()
+        # adapt the show_if-related fields to the newly generated ids
+        for new_item in item_map.values():
+            show_if_field_id = new_item.data.get('show_if_field_id')
+            if show_if_field_id is not None:
+                new_item.data['show_if_field_id'] = item_map[show_if_field_id].id
```

Comment:
@ThiefMaster I'm pretty sure you can have a clue on what's going on here. Turns out the item is not modified in the database, and maintains the old value despite being modified here. It doesn't work either if I call `db.session.add(new_item)`.

Reply:
If it was already flushed to the DB, you need to `flag_modified(new_item, 'data')` so SQLAlchemy knows it has to update the field in the DB.

`from sqlalchemy.orm.attributes import flag_modified` to import it

Reply:
Perfect, thanks a lot 🙏 

Addressed in [4128e93](https://github.com/indico/indico/pull/6678/commits/4128e93f9048005b920a4871430a08a504c9c340).